### PR TITLE
Added ImageFont.MAX_STRING_LENGTH

### PR DESCRIPTION
**Pull request by radarhere**

### CODE REVIEW

File: Tests/test_imagefont.py [hunk 1]
```diff
@@ -1038,6 +1038,25 @@ def test_render_mono_size():
...
+
+    transposed_font = ImageFont.TransposedFont(font)
+    with pytest.raises(ValueError):
+        transposed_font.getlength("A" * 1000001)
+
+    default_font = ImageFont.load_default()
+    with pytest.raises(ValueError):
+        default_font.getlength("A" * 1000001)
+    with pytest.raises(ValueError):
+        default_font.getbbox("A" * 1000001)
```
Comment (hugovk) [inline Tests/test_imagefont.py:1057]: Re: https://peps.python.org/pep-0515, underscores for readability: 


```suggestion
        font.getlength("A" * 1_000_001)
    with pytest.raises(ValueError):
        font.getbbox("A" * 1_000_001)
    with pytest.raises(ValueError):
        font.getmask2("A" * 1_000_001)

    transposed_font = ImageFont.TransposedFont(font)
    with pytest.raises(ValueError):
        transposed_font.getlength("A" * 1_000_001)

    default_font = ImageFont.load_default()
    with pytest.raises(ValueError):
        default_font.getlength("A" * 1_000_001)
    with pytest.raises(ValueError):
        default_font.getbbox("A" * 1_000_001)
```

File: src/PIL/ImageFont.py [hunk 2]
```diff
@@ -41,6 +41,9 @@ class Layout(IntEnum):
     RAQM = 1
 
 
+MAX_STRING_LENGTH = 1000000
```
Comment (hugovk) [inline src/PIL/ImageFont.py:44]:
```suggestion
MAX_STRING_LENGTH = 1_000_000
```

**Review by hugovk — APPROVED** [review]

**Comment by hugovk** [comment]
Noting this was assigned CVE-2023-44271.

https://nvd.nist.gov/vuln/detail/CVE-2023-44271

https://github.com/python-pillow/Pillow/pull/7520

**Comment by charlesoconor** [comment]
Would it be possible to release earlier versions with this cherry-picked on? Since it's a CVE.

**Comment by hugovk** [comment]
No, we only support the latest version, please upgrade or patch. 

This is a Python-only change (no C changes) so will be easier to patch if you can't upgrade.

And there have been more security fixes since, such as in this week's release. We do recommend upgrading.

https://pillow.readthedocs.io/en/stable/releasenotes/10.2.0.html#security

**Pull request merged**